patchwork-world-types/src/main/java/net/minecraftforge/common/extensions/IForgeWorldType.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Minecraft Forge, Patchwork Project
+ * Copyright (c) 2016-2020, 2019-2020
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.common.extensions;
+
+import java.util.function.LongFunction;
+
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.gui.screen.CustomizeBuffetLevelScreen;
+import net.minecraft.client.gui.screen.CustomizeFlatLevelScreen;
+import net.minecraft.client.gui.screen.world.CreateWorldScreen;
+import net.minecraft.world.IWorld;
+import net.minecraft.world.World;
+import net.minecraft.world.biome.layer.AddBambooJungleLayer;
+import net.minecraft.world.biome.layer.EaseBiomeEdgeLayer;
+import net.minecraft.world.biome.layer.ScaleLayer;
+import net.minecraft.world.biome.layer.SetBaseBiomesLayer;
+import net.minecraft.world.biome.layer.util.LayerFactory;
+import net.minecraft.world.biome.layer.util.LayerSampleContext;
+import net.minecraft.world.biome.layer.util.LayerSampler;
+import net.minecraft.world.gen.chunk.ChunkGenerator;
+import net.minecraft.world.gen.chunk.OverworldChunkGeneratorConfig;
+import net.minecraft.world.level.LevelGeneratorType;
+
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+
+import net.patchworkmc.impl.worldtypes.LevelGeneratorTypeFactory;
+import net.patchworkmc.mixin.worldtypes.AccessorBiomeLayers;
+
+public interface IForgeWorldType {
+	default LevelGeneratorType getWorldType() {
+		//return (LevelGeneratorType) this;
```

Comment:
I've left this line, but commented, as it's the original forge implementation. However the way I'm doing world types separates LevelGeneratorType from IForgeWorldType, so we use the below line I added.

I can remove this line if you want